io/zenoh-transport/src/multicast/manager.rs

```diff
@@ -170,9 +168,9 @@ impl TransportManager {
     pub async fn close_multicast(&self) {
         log::trace!("TransportManagerMulticast::clear())");
 
-        zlock!(self.state.multicast.protocols).clear();
+        zasynclock!(self.state.multicast.protocols).clear();
 
-        let mut tm_guard = zlock!(self.state.multicast.transports)
+        let mut tm_guard = zasynclock!(self.state.multicast.transports)
```

Comment:
alternatively, you could swap an empty vec with the transports, and then iterate over transports. Probably overkill since we're closing through, but I also think it'd be easier to understand.

Reply:
I've replaced with a single `drain()` iterator. I believe now it should be more compact and easier to read while keeping the lock as before.

Reply:
My point was that you could steal the Vec from the lock (steal 3 usizes), and then do the iteration after the lock is released

Reply:
Yes, but I'd rather guarantee there are no concurrency issues when closing multicast transports and someone, for any reason, still tries to open new multicast transports. In normal situations that would never happen but keeping the lock would keep us safe.